pkg/uefivars/boot/efiDppMessage.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package boot
+
+import (
+	"encoding/binary"
+	"fmt"
+)
+
+type EfiDppMsgSubType EfiDevPathProtoSubType
+
+const (
+	DppMsgTypeATAPI      EfiDppMsgSubType = iota + 1
+	DppMsgTypeSCSI                        //2
+	DppMsgTypeFibreCh                     //3
+	DppMsgTypeFirewire                    //4
+	DppMsgTypeUSB                         //5
+	DppMsgTypeIIO                         //6
+	_                                     //7
+	_                                     //8
+	DppMsgTypeInfiniband                  //9
+	DppMsgTypeVendor                      //10 //uart flow control, sas are subtypes
+	DppMsgTypeMAC                         //11
+	DppMsgTypeIP4                         //12
+	DppMsgTypeIP6                         //13
+	DppMsgTypeUART                        //14
+	DppMsgTypeUSBClass                    //15
+	DppMsgTypeUSBWWID                     //16
+	DppMsgTypeDevLU                       //17
+	DppMsgTypeSATA                        //18
+	DppMsgTypeISCSI                       //19
+	DppMsgTypeVLAN                        //20
+	_                                     //21
+	DppMsgTypeSASEx                       //22
+	DppMsgTypeNVME                        //23
+	DppMsgTypeURI                         //24
+	DppMsgTypeUFS                         //25
+	DppMsgTypeSD                          //26
+	DppMsgTypeBT                          //27
+	DppMsgTypeWiFi                        //28
+	DppMsgTypeeMMC                        //29
+	DppMsgTypeBLE                         //30
+	DppMsgTypeDNS                         //31
+	DppMsgTypeNVDIMM                      //32
+	DppMsgTypeRest                        //documented as 32, likely 33
+)
+
+func (e EfiDppMsgSubType) String() string {
+	switch e {
```

Comment:
could these stringers be maps to be shorter?

```go
var msgSubTypeStrings = map[EfiDppMsgSubType]string{
  DppMsgTypeATAPI: "ATAPI",
  ..
}
```

I'll leave deciding this up to you

Reply:
hadn't thought of that - much shorter.